pages/[user]/[type].tsx

```diff
@@ -24,52 +24,44 @@ export default function Type(props: InferGetServerSidePropsType<typeof getServer
 
   const { isReady } = Theme(props.user.theme);
 
-  const [selectedDate, setSelectedDate] = useState<Dayjs>(() => {
-    return props.date && dayjs(props.date).isValid() ? dayjs(props.date) : null;
-  });
   const [isTimeOptionsOpen, setIsTimeOptionsOpen] = useState(false);
   const [timeFormat, setTimeFormat] = useState("h:mma");
   const telemetry = useTelemetry();
 
+  const selectedDate = useMemo(() => {
+    if (typeof router.query.date !== "string") {
+      return null;
+    }
+    const parsed = dayjs(router.query.date);
+
+    return parsed.isValid() ? parsed : null;
+  }, [router.query.date]);
+
```

Comment:
cc @PeerRich this is what i mean with relying on router instead of `useState`

Reply:
agree